libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackClientWrapper.cs

```diff
@@ -0,0 +1,760 @@
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Bot.Schema;
+using Newtonsoft.Json;
+using SlackAPI;
+using SlackAPI.RPCMessages;
+
+using Attachment = SlackAPI.Attachment;
+
+namespace Microsoft.Bot.Builder.Adapters.Slack
+{
+    public class SlackClientWrapper
+    {
+        private const string PostMessageUrl = "https://slack.com/api/chat.postMessage";
+        private const string PostEphemeralMessageUrl = "https://slack.com/api/chat.postEphemeral";
+
+        private readonly SlackTaskClient _api;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SlackClientWrapper"/> class.
+        /// Creates a Slack client by supplying the access token.
+        /// </summary>
+        /// <param name="options">An object containing API credentials, a webhook verification token and other options.</param>
+        public SlackClientWrapper(SlackAdapterOptions options)
+        {
+            Options = options ?? throw new ArgumentNullException(nameof(options));
+
+            if (string.IsNullOrWhiteSpace(options.VerificationToken) && string.IsNullOrWhiteSpace(options.ClientSigningSecret))
+            {
+                const string warning = "****************************************************************************************" +
+                                       "* WARNING: Your bot is operating without recommended security mechanisms in place.     *" +
+                                       "* Initialize your adapter with a clientSigningSecret parameter to enable               *" +
+                                       "* verification that all incoming webhooks originate with Slack:                        *" +
+                                       "*                                                                                      *" +
+                                       "* var adapter = new SlackAdapter({clientSigningSecret: <my secret from slack>});       *" +
+                                       "*                                                                                      *" +
+                                       "****************************************************************************************" +
+                                       ">> Slack docs: https://api.slack.com/docs/verifying-requests-from-slack";
+
+                throw new Exception(warning + Environment.NewLine + "Required: include a verificationToken or clientSigningSecret to verify incoming Events API webhooks");
+            }
+
+            _api = new SlackTaskClient(options.BotToken);
+            LoginWithSlackAsync(default).Wait();
+        }
+
+        /// <summary>
+        /// Gets the SlackAdapterOptions.
+        /// </summary>
+        /// <value>
+        /// An object containing API credentials, a webhook verification token and other options.
+        /// </value>
+        public SlackAdapterOptions Options { get; private set; }
```

Comment:
private set; is not used. Can be removed.

Reply:
Done. Thanks!